From c84f4e9f13f99709ff83e9e982c78858bb4b0209 Mon Sep 17 00:00:00 2001
From: lawtlee <lawsac2947@gmail.com>
Date: Thu, 6 Jul 2023 09:19:21 -0700
Subject: [PATCH] Changed request_user.yml

Changed it to send_patch.yml as well as
changing functionality to sending a patch email
to me if there are any pull requests closing
---
 .github/workflows/request_user.yml | 41 ------------------------------
 .github/workflows/send_patch.yml   | 25 ++++++++++++++++++
 test.txt                           |  3 +++
 3 files changed, 28 insertions(+), 41 deletions(-)
 delete mode 100644 .github/workflows/request_user.yml
 create mode 100644 .github/workflows/send_patch.yml
 create mode 100644 test.txt

diff --git a/.github/workflows/request_user.yml b/.github/workflows/request_user.yml
deleted file mode 100644
index 53b9fbd..0000000
--- a/.github/workflows/request_user.yml
+++ /dev/null
@@ -1,41 +0,0 @@
-name: Request User on Pull Request
-on:
-  pull_request:
-    types:
-      - opened
-
-jobs:
-  request_user:
-    runs-on: ubuntu-latest
-    steps:
-      - name: Install Node.js
-        uses: actions/setup-node@v2
-        with:
-          node-version: '14'
-
-      - name: Request User
-        run: |
-          npm install @octokit/rest
-          node -e '
-            const { Octokit } = require("@octokit/rest");
-            const octokit = new Octokit({ auth: process.env.GITHUB_TOKEN });
-
-            const owner = process.env.GITHUB_REPOSITORY_OWNER;
-            const repo = process.env.GITHUB_REPOSITORY;
-            const pullRequestNumber = process.env.GITHUB_EVENT_PULL_REQUEST_NUMBER;
-            const requestedUser = "lawtlee";
-
-            octokit.issues.addAssignees({
-              owner: owner,
-              repo: repo,
-              issue_number: pullRequestNumber,
-              assignees: [requestedUser]
-            })
-            .then(() => {
-              console.log(`User ${requestedUser} requested for pull request #${pullRequestNumber}`);
-            })
-            .catch((error) => {
-              console.error(error);
-              process.exit(1);
-            });
-          '
\ No newline at end of file
diff --git a/.github/workflows/send_patch.yml b/.github/workflows/send_patch.yml
new file mode 100644
index 0000000..22412e6
--- /dev/null
+++ b/.github/workflows/send_patch.yml
@@ -0,0 +1,25 @@
+name: Request User on Pull Request
+on:
+  pull_request:
+    types:
+      - closed
+
+jobs:
+  email_patch:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v2
+      
+      - name: Create Patch
+        run: |
+          git format-patch main -1 HEAD
+      
+      - name: send email
+        run: |
+          patch=$(ls | grep *.patch)
+          git send-email --to=lawsac2947@gmail.com --from=lawsac2947@gmail.com
+
+      - name: Delete Patch
+        run: |
+          rm $(ls | grep *.patch)
diff --git a/test.txt b/test.txt
new file mode 100644
index 0000000..f5d3c89
--- /dev/null
+++ b/test.txt
@@ -0,0 +1,3 @@
+hello
+hello
+hello3
-- 
2.39.2 (Apple Git-143)

